packages/compose/src/hooks/use-debounce/index.js

```diff
@@ -0,0 +1,16 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+import { useMemoOne } from 'use-memo-one';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+export default function useDebounce( ...args ) {
+	const debounced = useMemoOne( () => debounce( ...args ), args );
+	useEffect( () => () => debounced.cancel(), [ debounced ] );
```

Comment:
`useMemoOne` seems preferable since we don't want have random cancellations. This also makes me wonder if the arguments should be "reactive" though. Maybe it's better to not allow a change of arguments so a previously debounced call is not accidentally cancelled. Not allowing a change of arguments is similar to binding it in a component constructor.